week03/find_the_index_of_the_first_occurrence_in_a_string.cpp

```diff
@@ -0,0 +1,25 @@
+class Solution {
+public:
+    int strStr(string haystack, string needle) {
+
+        for(int i=0; i<haystack.length(); i++){
+            if(haystack[i] == needle[0]){
+                int j = 0;
+                int startingIndex = i;
+
+                while(j<needle.length() && i<haystack.length() && haystack[i] == needle[j]){
+                    i++;
+                    j++;
+                }
+
+                if(j >= needle.length())return startingIndex;
+                i = startingIndex;
+
+            }
+        }
+
+        return -1;
+    }
+};
+
+/** This solution beats 100% runtime **/
```

Comment:
Doesn't matter. The only thing matters is time complexity and you haven't mentioned any.